pennylane/templates/subroutines/qft.py

```diff
@@ -115,4 +115,5 @@ def decomposition(wires):
         return decomp_ops
 
     def adjoint(self):
-        return QFT(wires=self.wires).inv()
+        self.queue()
+        return self.inv()
```

Comment:
Hi @KetpuntoG, just thought about this a bit more. It might likely be good to create a new object here.
```suggestion
        adjoint_op = QFT(wires=self.wires)
        adjoint_op.inverse = not self.inverse
        return adjoint_op
```
The reason is, that otherwise, we'll be referencing the same operation (`self`) that has already been created. The previous implementation created a new object, just set the inverse incorrectly.

An example that would likely lead to unexpected behaviour could for example be:
```python
import pennylane as qml

op = qml.QFT(wires=[0])
print(op.adjoint(), op)
```
```
QFT.inv(wires=[0]) QFT.inv(wires=[0])
```
With the suggested changes, this returns:
```
QFT.inv(wires=[0]) QFT(wires=[0])
```

Reply:
Also note, that by calling `QFT(wires=self.wires)`, queuing happens in the background already, hence we didn't require it previously.

Reply:
What you say have sense but in that case this example doesn't work
```python
dev = qml.device("default.qubit", wires=4)
@qml.qnode(dev)
def circuit():
    qml.adjoint(qml.QFT)(wires = range(4))
    return qml.probs(0)
```
The problem is that the information of the inverse is sometimes stored in the operator itself and sometimes only in the context producing some inconsistency sometimes. I think it is solved, these three cases work:

```python
dev = qml.device("default.qubit", wires=4)
@qml.qnode(dev)
def circuit():
    qml.QFT(wires = range(4))
    qml.adjoint(qml.QFT)(wires = range(4))
    qml.QFT(wires = range(4)).inv()
    return qml.probs(0)
```

and the other example too: `op = qml.QFT(wires=[0])
print(op.adjoint(), op)`

If it suits you, you can merge it directly 🙂

Reply:
in fact my change does not work either, I am still investigating.

Reply:
Solved! 🙌